client/components/site-selector/index.jsx

```diff
@@ -263,7 +293,7 @@ export default React.createClass( {
 					/>
 					<div className="site-selector__sites" ref="selector">
 						{ this.renderAllSites() }
-						{ this.renderRecentSites() }
+						{ this.renderStarredSites() }
```

Comment:
@aduth lifted the site count limit for starred sites. Mind giving another go for that user?